File: domain/src/main/java/ru/spliterash/spcore/domain/usecase/abstractCases/AbstractCheckEntityExistUseCase.java
package ru.spliterash.spcore.domain.usecase.abstractCases;

import lombok.RequiredArgsConstructor;
import ru.spliterash.spcore.domain.repo.CrudRepository;
import ru.spliterash.spcore.domain.usecase.UseCase;

@RequiredArgsConstructor
public abstract class AbstractCheckEntityExistUseCase implements UseCase<String, Boolean> {
    protected final CrudRepository<?, ?> repository;

    @Override
    public Boolean execute(String id) {
        return repository.exist(id);
    }
}

File: domain/src/main/java/ru/spliterash/spcore/domain/usecase/nonBlockingCases/AbstractCheckEntityExistUseCase.kt
package ru.spliterash.spcore.domain.usecase.nonBlockingCases

import lombok.RequiredArgsConstructor
import ru.spliterash.spcore.domain.repo.CrudRepository
import ru.spliterash.spcore.domain.usecase.NonBlockingUseCase

@RequiredArgsConstructor
abstract class AbstractCheckEntityExistUseCase(
    protected val repository: CrudRepository<*, *>
) : NonBlockingUseCase<String, Boolean> {
    override suspend fun execute(id: String): Boolean {
        return repository.exist(id)
    }
}